src/analysis/retail/evoker/devastation/modules/normalizers/EternitySurgeNormalizer.ts

```diff
@@ -0,0 +1,278 @@
+      if (!currentEternitySurgeCast) {
+        // Shouldn't be possible but just in case
+        return;
+      }
+
+      AddRelatedEvent(currentEternitySurgeCast, ETERNITY_SURGE_FROM_CAST, damageEvent);
+      AddRelatedEvent(damageEvent, ETERNITY_SURGE_FROM_CAST, currentEternitySurgeCast);
+    };
+
+    /** Figure out which damage window is from the cast
+     * @return the index of the window */
+    const findCastWindow = () => {
+      // 95% of the time we only have one damage window
+      if (damageWindows.length === 1) {
+        return 0;
+      }
+
+      /** We can have multiple damage windows in close proximity to each other
+       * We need to figure out which one is from the cast
+       * we do this by comparing the amount of damage in each window
+       * and choosing the one with the most damage, since Scintillation procs are
+       * 50% of base damage.
+       *
+       * It can also be determined by the amount of hits
+       * since procs can only ever hit a max of 1/2 targets depending on if
+       * Eternity Span is talented. */
+      const { windowIdxToUse } = damageWindows.reduce(
+        (acc, window, idx) => {
+          // We've already found the correct window, based on hit count
+          if (acc.maxWindowSize > hitsPerEmpowerRank) {
+            return acc;
+          }
+
+          // We hit more targets than is possible from procs so must be correct window
+          if (window.length > hitsPerEmpowerRank) {
+            acc.windowIdxToUse = idx;
+            acc.maxWindowSize = window.length;
+            return acc;
+          }
+
+          /** We'll just compare the first events in the windows.
+           * We could loop through all events in the windows and eg. compare
+           * the average value.
+           * But this works perfectly fine outside of some edge cases where the
+           * first hit might hit an immune / DR'ed target.
+           * But that would most likely be an issue event when looping.
+           *
+           * And for 95% of cases we only have one window to begin with, so this
+           * logic is already rarely used, so we'll just accept this potentiality. */
+          const initialEvent = window[0];
+          let rawAmount = initialEvent.amount + (initialEvent.absorbed ?? 0);
+          let normalizedAmount = 0;
+
+          /** If it's a crit we need to normalize the value, a crit normally means double amount,
+           * but we also have a talent that increases it 30%, so we need to decrease that amount too */
+          const isCrit = initialEvent.hitType === HIT_TYPES.CRIT;
+          if (isCrit) {
+            if (hasSpellweaversDominance) {
+              rawAmount -= calculateEffectiveDamageFromCritDamageIncrease(
+                initialEvent,
+                SPELLWEAVERS_DOMINANCE_CRIT_MULTIPLIER,
+              );
+            }
+
+            rawAmount /= 2;
+          }
+
+          normalizedAmount = rawAmount;
```

Comment:
minor nit: `rawAmount` doesn't need to be a separate variable, right?

Reply:
Yea that is a leftover var from previous version of that calc 😅